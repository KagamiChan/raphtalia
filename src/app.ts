import Fastify, { FastifyInstance } from 'fastify'

export const buildFastify = (): FastifyInstance => {
  const app = Fastify()

  app.post('/webhook', (request, reply) => {
    reply.send('hello world')
  })

  return app
}
