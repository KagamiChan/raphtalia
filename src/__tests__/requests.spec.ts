import request from 'supertest'
import glob from 'glob'
import { each } from 'lodash'
import path from 'path'

import { app } from '../app'

describe('webhook events', () => {
  const events = glob.sync(path.resolve(__dirname, './events/*.json'))
  each(events, event => {
    const eventName = path.basename(event).replace('.json', '')
    return it(eventName, async done => {
      const resp = await request(app.callback())
        .post('/app')
        .set('Accept', 'application/json')
        .set('X-GitHub-Event', eventName)
        .expect(200)

      expect(resp.body).toMatchSnapshot()
      done()
    })
  })
})
