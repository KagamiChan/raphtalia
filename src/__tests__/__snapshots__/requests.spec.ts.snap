// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webhook events commit_comment 1`] = `Object {}`;

exports[`webhook events issue_comment 1`] = `Object {}`;

exports[`webhook events issues 1`] = `Object {}`;

exports[`webhook events pull_request 1`] = `Object {}`;

exports[`webhook events push 1`] = `Object {}`;
